thingif/src/main/java/com/kii/thingif/ThingIFAPI.java

```diff
@@ -253,7 +255,23 @@ public Builder registerAction(
                 @NonNull String alias,
                 @NonNull String actionName,
                 @NonNull Class<? extends Action> actionClass){
-            //TODO: validate actionClass, should contain only 1 field
+            // validate field of actionClass
+            List<Field> fields = new ArrayList<>();
+            for (Field field : actionClass.getDeclaredFields()) {
+                if (!field.getName().equals("this$0")){ // in case of inner class, has this field
```

Comment:
@yongpingchen 

Can we use [isLocalClass()](https://developer.android.com/reference/java/lang/Class.html#isLocalClass()) and/or [isMemberClass()](https://developer.android.com/reference/java/lang/Class.html#isMemberClass())?

I'm not sure above 2 methods satisfies your requirement or not, but, I think it is indirect way to check a class is inner class or not with `Field`.

What do you think?